tools/c7n_azure/c7n_azure/resources/storage.py

```diff
@@ -550,3 +550,64 @@ def _process_resource(self, resource):
             resource['name'],
             StorageAccountUpdateParameters(enable_https_traffic_only=self.data.get('value'))
         )
+
+
+@Storage.filter_registry.register('soft-delete')
+class SoftDeleteFilter(Filter):
+    """
+    Filter by the current soft delete
+    configuration for this storage account.
+    :example:
+    Find storage accounts with blob services soft delete disabled
+    .. code-block:: yaml
+        policies:
+          - name: storage-no-soft-delete
+            resource: azure.storage
+            filters:
+              - type: soft-delete
+                enabled: false
+    """
+
+    schema = type_schema(
+        'soft-delete',
+        required=['type', 'enabled'],
+        **{
+            'enabled': {"type": "boolean"},
+        }
+    )
+
+    log = logging.getLogger('custodian.azure.storage.soft-delete-filter')
+
+    def __init__(self, data, manager=None):
+        super(SoftDeleteFilter, self).__init__(data, manager)
+        self.enabled = self.data['enabled']
+
+    def process(self, resources, event=None):
+        resources, exceptions = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._process_resource_set,
+            executor_factory=self.executor_factory,
+            log=self.log
+        )
+        if exceptions:
+            raise exceptions[0]
+        return resources
+
+    def _process_resource_set(self, resources, event=None):
+        client = self.manager.get_client()
+        result = []
+        for resource in resources:
+            if 'blobServices' not in resource['properties']:
```

Comment:
we typically augment with a namespaced annotation key (c7n:blob-services) versus introducing new properties as though they were native. the azure provider does appear to have both (c7n: prefix annotations) and some legacy direct property mutation, although the later is limited.

Reply:
updated to use `c7n` prefix